week3/mission1/src/App.tsx

```diff
@@ -0,0 +1,487 @@
+  const go = (p: number) => setSp({ page: String(Math.max(1, p)) });
+
+  return (
+    <div className="mx-auto max-w-6xl px-4 pt-2 pb-4">
+      <div className="flex items-center justify-center gap-2">
+        <button
+          onClick={() => go(page - 1)}
+          disabled={page <= 1}
+          className="px-3 py-1 rounded-md border border-white/20 disabled:opacity-40"
+        >
+          ◀
+        </button>
+        <span className="text-sm text-white/80">{page} 페이지</span>
+        <button
+          onClick={() => go(page + 1)}
+          className="px-3 py-1 rounded-md border border-white/20"
+        >
+          ▶
+        </button>
+      </div>
+    </div>
+  );
+}
+
+// 레이아웃
+function Layout() {
+  const location = useLocation();
+  const isDetail = location.pathname.startsWith("/movie/");
+  return (
+    <div className="min-h-screen bg-neutral-900 text-white">
+      <header className="sticky top-0 z-10 bg-neutral-900/80 backdrop-blur border-b border-white/10">
+        <div className="mx-auto max-w-6xl px-4 py-4 flex items-center justify-between"></div>
+        <div className="mx-auto max-w-6xl px-4">
+          <Navbar />
+        </div>
+        {!isDetail && <TopPager />}
+      </header>
+
+      <main className="mx-auto max-w-6xl px-4 py-6">
+        <Outlet />
+      </main>
+    </div>
+  );
+}
+
+// 목록페이지
+function MoviesPage({ category }: { category: Category }) {
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<unknown>(null);
+
+  const [sp] = useSearchParams();
+  const page = Math.max(1, Number(sp.get("page") ?? 1));
+
+  useEffect(() => {
+    let aborted = false;
+    (async () => {
+      try {
+        setLoading(true);
+        setError(null);
+        const res = await TMDB.get(`/movie/${category}`, {
+          params: { language: "ko-KR", page },
+        });
+        if (!aborted) setMovies(res.data.results ?? []);
+      } catch (e) {
+        if (!aborted) setError(e);
+      } finally {
+        if (!aborted) setLoading(false);
+      }
+    })();
+    return () => {
+      aborted = true;
+    };
+  }, [category, page]);
+
+  if (loading) return <Loader />;
+
+  if (error)
+    return (
+      <div className="min-h-[40vh] grid place-items-center text-red-500">
+        데이터를 불러오지 못했습니다.
+      </div>
+    );
+
+  return (
+    <ul
+      className="
+        grid gap-4
+        grid-cols-2 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-5 xl:grid-cols-6
+      "
+    >
+      {movies.map((m) => (
+        <li key={m.id}>
+          <Link to={`/movie/${m.id}`} className="block">
+            <MovieCard
+              title={m.title ?? m.name}
+              posterPath={m.poster_path ?? null}
+              vote={m.vote_average}
+              release={m.release_date ?? m.first_air_date}
+            />
+          </Link>
+        </li>
+      ))}
+    </ul>
+  );
```

Comment:
저는 개인적으로 이 부분에서는 조건 연산자를 활용하는 게 좋다고 생각합니다.
```suggestion
return loading ? (
  <Loader />
) : error ? (
  <div className="min-h-[40vh] grid place-items-center text-red-500">
    데이터를 불러오지 못했습니다.
  </div>
) : (
  <ul
    className="
      grid gap-4
      grid-cols-2 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-5 xl:grid-cols-6
    "
  >
    {movies.map((m) => (
      <li key={m.id}>
        <Link to={`/movie/${m.id}`} className="block">
          <MovieCard
            title={m.title ?? m.name}
            posterPath={m.poster_path ?? null}
            vote={m.vote_average}
            release={m.release_date ?? m.first_air_date}
          />
        </Link>
      </li>
    ))}
  </ul>
);
```